From c2dda4ff0a6456f2b30406c9b5a7024a2f7c474d Mon Sep 17 00:00:00 2001
From: aleks <aleks.stier@icloud.com>
Date: Wed, 27 May 2020 17:57:15 +0200
Subject: [PATCH] Apply netwmicon-0.8.4

Enables to set *_NET_WM_ICON* which hardcodes an icon for st. An icon is
already defined in the file *icon.h* which is stolen from
[flat-remix](https://github.com/daniruiz/flat-remix).

Generally the icon of an application is defined by its desktop-entry. A patch
with the name [desktopentry](../desktopentry) already exists for this purpose.
However, some programs like tint2 do not respect the desktopentry and rely
instead on an hardcoded icon which has to be defined by *_NET_WM_ICON*. Since
st does not define *_NET_WM_ICON* those programs will display some default icon
(which is ugly).

You can of course change the icon to any icon you want. Just grab some icon
from your favorite icon-theme. The tricky part is that it needs to be encoded
as *"an array of 32bit packed CARDINAL ARGB with high byte being A, low byte
being B" -
[Source](https://specifications.freedesktop.org/wm-spec/1.3/ar01s05.html)*.
This can be done with the script [netwmicon.sh](netwmicon.sh). It takes as
argument the icon-file and prints to stdout the encoded icon. Redirect it to
icon.h to save it and reinstall st. You need to install both *imagemagick* and
*inkscape* for the script to work.

---
 icon.h | 686 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 x.c    |   7 +-
 2 files changed, 692 insertions(+), 1 deletion(-)
 create mode 100644 icon.h

diff --git a/icon.h b/icon.h
new file mode 100644
index 0000000..7ab09d0
--- /dev/null
+++ b/icon.h
@@ -0,0 +1,686 @@
+unsigned long icon[] = {
+	64, 64,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x02000000, 0x03000000,
+	0x03000000, 0x03000000, 0x03000000, 0x03000000, 0x03000000, 0x03000000,
+	0x03000000, 0x03000000, 0x03000000, 0x03000000, 0x03000000, 0x03000000,
+	0x03000000, 0x03000000, 0x03000000, 0x03000000, 0x03000000, 0x03000000,
+	0x03000000, 0x03000000, 0x03000000, 0x03000000, 0x03000000, 0x03000000,
+	0x03000000, 0x03000000, 0x03000000, 0x03000000, 0x03000000, 0x03000000,
+	0x03000000, 0x03000000, 0x03000000, 0x03000000, 0x03000000, 0x03000000,
+	0x03000000, 0x03000000, 0x03000000, 0x03000000, 0x03000000, 0x03000000,
+	0x03000000, 0x03000000, 0x03000000, 0x03000000, 0x03000000, 0x03000000,
+	0x03000000, 0x02000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x03000000,
+	0x20181818, 0x4e868686, 0x74b2b2b2, 0x77b6b6b6, 0x77b6b6b6, 0x77b6b6b6,
+	0x77b6b6b6, 0x77b6b6b6, 0x77b6b6b6, 0x77b6b6b6, 0x77b6b6b6, 0x77b6b6b6,
+	0x77b6b6b6, 0x77b6b6b6, 0x77b6b6b6, 0x77b6b6b6, 0x77b6b6b6, 0x77b6b6b6,
+	0x77b6b6b6, 0x77b6b6b6, 0x77b6b6b6, 0x77b6b6b6, 0x77b6b6b6, 0x77b6b6b6,
+	0x77b6b6b6, 0x77b6b6b6, 0x77b6b6b6, 0x77b6b6b6, 0x77b6b6b6, 0x77b6b6b6,
+	0x77b6b6b6, 0x77b6b6b6, 0x77b6b6b6, 0x77b6b6b6, 0x77b6b6b6, 0x77b6b6b6,
+	0x77b6b6b6, 0x77b6b6b6, 0x77b6b6b6, 0x77b6b6b6, 0x77b6b6b6, 0x77b6b6b6,
+	0x77b6b6b6, 0x77b6b6b6, 0x77b6b6b6, 0x77b6b6b6, 0x77b6b6b6, 0x77b6b6b6,
+	0x77b6b6b6, 0x77b6b6b6, 0x77b6b6b6, 0x77b6b6b6, 0x77b6b6b6, 0x74b2b2b2,
+	0x4e868686, 0x20181818, 0x03000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x03000000, 0x46717171, 0xcef3f3f3, 0xffffffff,
+	0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff,
+	0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff,
+	0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff,
+	0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff,
+	0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff,
+	0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff,
+	0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff,
+	0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff,
+	0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xcdf3f3f3,
+	0x456f6f6f, 0x03000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x211f1f1f, 0xd1f4f4f4, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff,
+	0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff,
+	0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff,
+	0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff,
+	0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff,
+	0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff,
+	0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff,
+	0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff,
+	0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff,
+	0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xd0f3f3f3, 0x20181818,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x59959595, 0xffffffff,
+	0xffffffff, 0xff8b8b8b, 0xff363636, 0xff363636, 0xff363636, 0xff363636,
+	0xff363636, 0xff363636, 0xff363636, 0xff363636, 0xff363636, 0xff363636,
+	0xff363636, 0xff363636, 0xff363636, 0xff363636, 0xff363636, 0xff363636,
+	0xff363636, 0xff363636, 0xff363636, 0xff363636, 0xff363636, 0xff363636,
+	0xff363636, 0xff363636, 0xff363636, 0xff363636, 0xff363636, 0xff363636,
+	0xff363636, 0xff363636, 0xff363636, 0xff363636, 0xff363636, 0xff363636,
+	0xff363636, 0xff363636, 0xff363636, 0xff363636, 0xff363636, 0xff363636,
+	0xff363636, 0xff363636, 0xff363636, 0xff363636, 0xff363636, 0xff363636,
+	0xff363636, 0xff363636, 0xff363636, 0xff363636, 0xff363636, 0xff363636,
+	0xff8c8c8c, 0xffffffff, 0xffffffff, 0x58919191, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x83b3b3b3, 0xffffffff, 0xffffffff, 0xff262626,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff262626, 0xffffffff,
+	0xffffffff, 0x83b3b3b3, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x8eacacac, 0xffffffff, 0xffffffff, 0xff222222, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff222222, 0xffffffff, 0xffffffff, 0x8eacacac,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x8eacacac, 0xffffffff,
+	0xffffffff, 0xff222222, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff222222, 0xffffffff, 0xffffffff, 0x8eacacac, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x8eacacac, 0xffffffff, 0xffffffff, 0xff222222,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff222222, 0xffffffff,
+	0xffffffff, 0x8eacacac, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x8eacacac, 0xffffffff, 0xffffffff, 0xff222222, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff222222, 0xffffffff, 0xffffffff, 0x8eacacac,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x8eacacac, 0xffffffff,
+	0xffffffff, 0xff222222, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff2c2c2c, 0xffe0e0e0, 0xff1c1c1c, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff222222, 0xffffffff, 0xffffffff, 0x8eacacac, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x8eacacac, 0xffffffff, 0xffffffff, 0xff222222,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff202020,
+	0xff6c6c6c, 0xffffffff, 0xff6d6d6d, 0xff3c3c3c, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff222222, 0xffffffff,
+	0xffffffff, 0x8eacacac, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x8eacacac, 0xffffffff, 0xffffffff, 0xff222222, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff2d2d2d, 0xffe1e1e1, 0xffc3c3c3, 0xffffffff,
+	0xffa1a1a1, 0xffdddddd, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff222222, 0xffffffff, 0xffffffff, 0x8eacacac,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x8eacacac, 0xffffffff,
+	0xffffffff, 0xff222222, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff7f7f7f, 0xffbfbfbf, 0xff303030, 0xffffffff, 0xff1c1c1c, 0xff181818,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff222222, 0xffffffff, 0xffffffff, 0x8eacacac, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x8eacacac, 0xffffffff, 0xffffffff, 0xff222222,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff717171, 0xffe1e1e1,
+	0xff545454, 0xffffffff, 0xff1c1c1c, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff222222, 0xffffffff,
+	0xffffffff, 0x8eacacac, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x8eacacac, 0xffffffff, 0xffffffff, 0xff222222, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff1c1c1c, 0xffa1a1a1, 0xfff6f6f6, 0xffffffff,
+	0xffaeaeae, 0xff515151, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff222222, 0xffffffff, 0xffffffff, 0x8eacacac,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x8eacacac, 0xffffffff,
+	0xffffffff, 0xff222222, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff343434, 0xffffffff, 0xff979797, 0xfff9f9f9,
+	0xff515151, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff222222, 0xffffffff, 0xffffffff, 0x8eacacac, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x8eacacac, 0xffffffff, 0xffffffff, 0xff222222,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff303030, 0xffffffff, 0xff1c1c1c, 0xffb3b3b3, 0xff8d8d8d, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff222222, 0xffffffff,
+	0xffffffff, 0x8eacacac, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x8eacacac, 0xffffffff, 0xffffffff, 0xff222222, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff686868, 0xff616161, 0xff3c3c3c, 0xffffffff,
+	0xff545454, 0xffe8e8e8, 0xff5b5b5b, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff222222, 0xffffffff, 0xffffffff, 0x8eacacac,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x8eacacac, 0xffffffff,
+	0xffffffff, 0xff222222, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff4b4b4b, 0xffb3b3b3, 0xffe1e1e1, 0xffffffff, 0xffcccccc, 0xff717171,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff222222, 0xffffffff, 0xffffffff, 0x8eacacac, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x8eacacac, 0xffffffff, 0xffffffff, 0xff222222,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff303030, 0xffffffff, 0xff1c1c1c, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff222222, 0xffffffff,
+	0xffffffff, 0x8eacacac, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x8eacacac, 0xffffffff, 0xffffffff, 0xff222222, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff2b2b2b, 0xffd1d1d1,
+	0xff1c1c1c, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff222222, 0xffffffff, 0xffffffff, 0x8eacacac,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x8eacacac, 0xffffffff,
+	0xffffffff, 0xff222222, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff7f7f7f, 0xffe9e9e9, 0xffe9e9e9, 0xffe9e9e9, 0xffe9e9e9,
+	0xffe9e9e9, 0xffe9e9e9, 0xff444444, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff222222, 0xffffffff, 0xffffffff, 0x8eacacac, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x8eacacac, 0xffffffff, 0xffffffff, 0xff222222,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff222222, 0xffffffff,
+	0xffffffff, 0x8eacacac, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x8eacacac, 0xffffffff, 0xffffffff, 0xff222222, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff222222, 0xffffffff, 0xffffffff, 0x8eacacac,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x8eacacac, 0xffffffff,
+	0xffffffff, 0xff222222, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff222222, 0xffffffff, 0xffffffff, 0x8eacacac, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x8eacacac, 0xffffffff, 0xffffffff, 0xff222222,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff222222, 0xffffffff,
+	0xffffffff, 0x8eacacac, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x8eacacac, 0xffffffff, 0xffffffff, 0xff222222, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff222222, 0xffffffff, 0xffffffff, 0x8eacacac,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x8eacacac, 0xffffffff,
+	0xffffffff, 0xff222222, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff222222, 0xffffffff, 0xffffffff, 0x8eacacac, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x8eacacac, 0xffffffff, 0xffffffff, 0xff222222,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff222222, 0xffffffff,
+	0xffffffff, 0x8eacacac, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x8eacacac, 0xffffffff, 0xffffffff, 0xff222222, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff222222, 0xffffffff, 0xffffffff, 0x8eacacac,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x8eacacac, 0xffffffff,
+	0xffffffff, 0xff222222, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff222222, 0xffffffff, 0xffffffff, 0x8eacacac, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x8eacacac, 0xffffffff, 0xffffffff, 0xff222222,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff222222, 0xffffffff,
+	0xffffffff, 0x8eacacac, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x8eacacac, 0xffffffff, 0xffffffff, 0xff222222, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff222222, 0xffffffff, 0xffffffff, 0x8eacacac,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x8eacacac, 0xffffffff,
+	0xffffffff, 0xff222222, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff222222, 0xffffffff, 0xffffffff, 0x8eacacac, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x8eacacac, 0xffffffff, 0xffffffff, 0xff222222,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff222222, 0xffffffff,
+	0xffffffff, 0x8eacacac, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x8eacacac, 0xffffffff, 0xffffffff, 0xff222222, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff222222, 0xffffffff, 0xffffffff, 0x8eacacac,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x8eacacac, 0xffffffff,
+	0xffffffff, 0xff222222, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff222222, 0xffffffff, 0xffffffff, 0x8eacacac, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x8eacacac, 0xffffffff, 0xffffffff, 0xff222222,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff222222, 0xffffffff,
+	0xffffffff, 0x8eacacac, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x8eacacac, 0xffffffff, 0xffffffff, 0xff222222, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff222222, 0xffffffff, 0xffffffff, 0x8eacacac,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x8eacacac, 0xffffffff,
+	0xffffffff, 0xff222222, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff222222, 0xffffffff, 0xffffffff, 0x8eacacac, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x8eacacac, 0xffffffff, 0xffffffff, 0xff222222,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff222222, 0xffffffff,
+	0xffffffff, 0x8eacacac, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x8eacacac, 0xffffffff, 0xffffffff, 0xff222222, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff222222, 0xffffffff, 0xffffffff, 0x8eacacac,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x8eacacac, 0xffffffff,
+	0xffffffff, 0xff222222, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff222222, 0xffffffff, 0xffffffff, 0x8eacacac, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x8aacacac, 0xffffffff, 0xffffffff, 0xff262626,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff171717,
+	0xff171717, 0xff171717, 0xff171717, 0xff171717, 0xff262626, 0xffffffff,
+	0xffffffff, 0x8aacacac, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x62858585, 0xffffffff, 0xffffffff, 0xff8c8c8c, 0xff373737, 0xff363636,
+	0xff363636, 0xff363636, 0xff363636, 0xff363636, 0xff363636, 0xff363636,
+	0xff363636, 0xff363636, 0xff363636, 0xff363636, 0xff363636, 0xff363636,
+	0xff363636, 0xff363636, 0xff363636, 0xff363636, 0xff363636, 0xff363636,
+	0xff363636, 0xff363636, 0xff363636, 0xff363636, 0xff363636, 0xff363636,
+	0xff363636, 0xff363636, 0xff363636, 0xff363636, 0xff363636, 0xff363636,
+	0xff363636, 0xff363636, 0xff363636, 0xff363636, 0xff363636, 0xff363636,
+	0xff363636, 0xff363636, 0xff363636, 0xff363636, 0xff363636, 0xff363636,
+	0xff363636, 0xff363636, 0xff363636, 0xff363636, 0xff363636, 0xff363636,
+	0xff363636, 0xff373737, 0xff8d8d8d, 0xffffffff, 0xffffffff, 0x62828282,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x21171717, 0xdee2e2e2,
+	0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff,
+	0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff,
+	0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff,
+	0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff,
+	0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff,
+	0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff,
+	0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff,
+	0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff,
+	0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff,
+	0xffffffff, 0xffffffff, 0xdee1e1e1, 0x1f101010, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x03000000, 0x5f4e4e4e, 0xdbe1e1e1, 0xffffffff,
+	0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff,
+	0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff,
+	0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff,
+	0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff,
+	0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff,
+	0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff,
+	0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff,
+	0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff,
+	0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xdae1e1e1,
+	0x5f4b4b4b, 0x03000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x12000000, 0x48040404, 0x6d595959, 0x8d929292, 0x90979797,
+	0x90979797, 0x90979797, 0x90979797, 0x90979797, 0x90979797, 0x90979797,
+	0x90979797, 0x90979797, 0x90979797, 0x90979797, 0x90979797, 0x90979797,
+	0x90979797, 0x90979797, 0x90979797, 0x90979797, 0x90979797, 0x90979797,
+	0x90979797, 0x90979797, 0x90979797, 0x90979797, 0x90979797, 0x90979797,
+	0x90979797, 0x90979797, 0x90979797, 0x90979797, 0x90979797, 0x90979797,
+	0x90979797, 0x90979797, 0x90979797, 0x90979797, 0x90979797, 0x90979797,
+	0x90979797, 0x90979797, 0x90979797, 0x90979797, 0x90979797, 0x90979797,
+	0x90979797, 0x90979797, 0x90979797, 0x90979797, 0x90979797, 0x90979797,
+	0x90979797, 0x8d929292, 0x6d595959, 0x48040404, 0x12000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x06000000, 0x22000000, 0x29000000, 0x29000000, 0x29000000, 0x29000000,
+	0x29000000, 0x29000000, 0x29000000, 0x29000000, 0x29000000, 0x29000000,
+	0x29000000, 0x29000000, 0x29000000, 0x29000000, 0x29000000, 0x29000000,
+	0x29000000, 0x29000000, 0x29000000, 0x29000000, 0x29000000, 0x29000000,
+	0x29000000, 0x29000000, 0x29000000, 0x29000000, 0x29000000, 0x29000000,
+	0x29000000, 0x29000000, 0x29000000, 0x29000000, 0x29000000, 0x29000000,
+	0x29000000, 0x29000000, 0x29000000, 0x29000000, 0x29000000, 0x29000000,
+	0x29000000, 0x29000000, 0x29000000, 0x29000000, 0x29000000, 0x29000000,
+	0x29000000, 0x29000000, 0x29000000, 0x29000000, 0x29000000, 0x29000000,
+	0x22000000, 0x06000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+	0x00000000, 0x00000000, 0x00000000, 0x00000000,
+};
diff --git a/x.c b/x.c
index 210f184..396ac33 100644
--- a/x.c
+++ b/x.c
@@ -17,6 +17,7 @@
 
 char *argv0;
 #include "arg.h"
+#include "icon.h"
 #include "st.h"
 #include "win.h"
 
@@ -94,7 +95,7 @@ typedef struct {
 	Window win;
 	Drawable buf;
 	GlyphFontSpec *specbuf; /* font spec buffer used for rendering */
-	Atom xembed, wmdeletewin, netwmname, netwmiconname, netwmpid;
+	Atom xembed, wmdeletewin, netwmname, netwmicon, netwmiconname, netwmpid;
 	struct {
 		XIM xim;
 		XIC xic;
@@ -1157,6 +1158,10 @@ xinit(int cols, int rows)
 	xw.netwmname = XInternAtom(xw.dpy, "_NET_WM_NAME", False);
 	xw.netwmiconname = XInternAtom(xw.dpy, "_NET_WM_ICON_NAME", False);
 	XSetWMProtocols(xw.dpy, xw.win, &xw.wmdeletewin, 1);
+	xw.netwmicon = XInternAtom(xw.dpy, "_NET_WM_ICON", False);
+	XChangeProperty(xw.dpy, xw.win, xw.netwmicon, XA_CARDINAL, 32,
+			PropModeReplace, (uchar *)&icon, LEN(icon));
+
 
 	xw.netwmpid = XInternAtom(xw.dpy, "_NET_WM_PID", False);
 	XChangeProperty(xw.dpy, xw.win, xw.netwmpid, XA_CARDINAL, 32,
-- 
2.27.0

